### PR TITLE
Fix tests for Python 3.11, don't emit `super-init-not-called` for `Enum` subclasses

### PR DESCRIPTION
**Pull request by AdamWill**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

This fixes some issues in the tests with Python 3.11, and makes one change to the actual checks that is exposed by Python 3.11 - not emitting `super-init-not-called` for subclasses of `Enum`. `Enum` has a `__init__` in Python 3.11, but it does nothing at all (just `pass`), and the examples in Python's own `Enum` docs include sample subclasses with their own `__init__` which do not call `Enum.__init__`, so clearly it's OK/intended for you to do that. I've [asked upstream why this useless `Enum.__init__` was added at all](https://github.com/python/cpython/pull/30582#issuecomment-1181055042).

### CODE REVIEW

**Comment by AdamWill** [comment]
Note, this is my first time touching pylint so please review carefully! There is still a problem with the `syntax_error` and `import_error` tests, discussed somewhat at https://github.com/PyCQA/pylint/pull/6551#issuecomment-1180893393 and replies - I wasn't able to come up with a sensibly upstreamable fix for that.

File: tests/functional/i/iterable_context_asyncio.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+"""
+Checks that we don't erroneously emit not-an-iterable errors for
+coroutines built with asyncio.coroutine.
```
Comment (DanielNoord) [inline tests/functional/i/iterable_context_asyncio.py:3]:
```suggestion
coroutines built with asyncio.coroutine.

These decorators were deprecated in 3.8 and removed in 3.10.
```

File: tests/functional/a/access/access_attr_before_def_false_positive.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # pylint: disable=invalid-name,too-many-public-methods,attribute-defined-outside-init
-# pylint: disable=useless-object-inheritance,too-few-public-methods
+# pylint: disable=useless-object-inheritance,too-few-public-methods deprecated-module
```
Comment (DanielNoord) [inline tests/functional/a/access/access_attr_before_def_false_positive.py:2]:
```suggestion
# pylint: disable=useless-object-inheritance,too-few-public-methods, deprecated-module
```

File: doc/whatsnew/2/2.15/index.rst [hunk 3]
```diff
@@ -33,6 +33,8 @@ False positives fixed
 * Don't report ``unsupported-binary-operation`` on Python <= 3.9 when using the ``|`` operator
   with types, if one has a metaclass that overloads ``__or__`` or ``__ror__`` as appropriate.
 
+* Don't report ``super-init-not-called`` for subclasses of ``Enum`` (on Python >= 3.11).
```
Comment (DanielNoord) [inline doc/whatsnew/2/2.15/index.rst:36]: This should probably be moved so the `Closes` is not associated with it.

**Review by DanielNoord — CHANGES_REQUESTED** [review]
Let's wait for a response on the `CPython` issue. I'd like to know if this is a conscious decision.

File: doc/whatsnew/2/2.15/index.rst [hunk 4]
```diff
@@ -33,6 +33,8 @@ False positives fixed
 * Don't report ``unsupported-binary-operation`` on Python <= 3.9 when using the ``|`` operator
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2/2.15/index.rst:33]:
```suggestion
* Don't report ``super-init-not-called`` for subclasses of ``Enum`` (on Python >= 3.11).


* Don't report ``unsupported-binary-operation`` on Python <= 3.9 when using the ``|`` operator
```

File: doc/whatsnew/2/2.15/index.rst [hunk 5]
```diff
@@ -33,6 +33,8 @@ False positives fixed
...
   with types, if one has a metaclass that overloads ``__or__`` or ``__ror__`` as appropriate.
 
+* Don't report ``super-init-not-called`` for subclasses of ``Enum`` (on Python >= 3.11).
+
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2/2.15/index.rst:37]:
```suggestion
```

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Comment by AdamWill** [comment]
I've re-factored the review changes into the original commits, and added a couple more fixes, including one for the syntax-error problem. In https://github.com/PyCQA/pylint/pull/6551 @DanielNoord suggested moving that test into the unit tests, but this just seems easier, at least until they change the description of *this* syntax error too...

**Comment by AdamWill** [comment]
The CI failures look to be problems in the CI infra, not in the PR.

**Comment by AdamWill** [comment]
@DanielNoord done, can you clear the 'change requested'?

File: tests/functional/i/iterable_context_asyncio.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+These decorators were deprecated in 3.8 and removed in 3.10.
```
Comment (DanielNoord) [inline tests/functional/i/iterable_context_asyncio.py:5]:
```suggestion
These decorators were deprecated in 3.8 and removed in 3.11.
```

**Review by DanielNoord — APPROVED** [review]
Made a mistake in my previous comment... 😅 

Thanks @AdamWill

**Pull request merged**